src/lib.rs

```diff
@@ -84,8 +85,8 @@ pub trait Driver {
     /// The url used to connect to this driver
     fn url(&self) -> &str;
     /// Start a session for this driver
-    fn session(self) -> Result<DriverSession, Error> where Self : Sized + 'static {
-        DriverSession::create_session(Box::new(self))
+    fn session(self, params: &NewSessionCmd) -> Result<DriverSession, Error> where Self : Sized + 'static {
```

Comment:
This is a (worthwhile) breaking change, so this release will need to be published as v0.2.0.